integrations/oci/objectstorage-health/src/main/java/io/helidon/integrations/oci/objectstorage/health/OciObjectStorageHealthCheck.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.integrations.oci.objectstorage.health;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import io.helidon.common.reactive.Single;
+import io.helidon.config.Config;
+import io.helidon.health.common.BuiltInHealthCheck;
+import io.helidon.integrations.common.rest.ApiOptionalResponse;
+import io.helidon.integrations.oci.objectstorage.GetBucket;
+import io.helidon.integrations.oci.objectstorage.GetBucketRx;
+import io.helidon.integrations.oci.objectstorage.OciObjectStorageRx;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Liveness;
+
+import static io.helidon.common.http.Http.Status.OK_200;
+import static java.time.temporal.ChronoUnit.MILLIS;
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+/**
+ * Liveness check for an OCI's ObjectStorage bucket. Reads configuration to
+ * obtain bucket name as well as OCI properties from '~/.oci/config'.
+ */
+@Liveness
+@ApplicationScoped // this will be ignored if not within CDI
+@BuiltInHealthCheck
+public final class OciObjectStorageHealthCheck implements HealthCheck {
+    private static final Logger LOGGER = Logger.getLogger(OciObjectStorageHealthCheck.class.getName());
+
+    @Inject
+    @ConfigProperty(name = "oci.objectstorage.bucket")
+    private String bucket;
+
+    @Inject
+    @ConfigProperty(name = "oci.objectstorage.namespace")
+    private String namespace;
+
+    @Inject
+    @ConfigProperty(name = "oci.objectstorage.health.timeout-millis", defaultValue = "10000")
+    private long timeoutMillis;
+
+    private Duration timeout;
+
+    private OciObjectStorageRx ociObjectStorage;
```

Comment:
All fields should be final. You can use OciObjectStoraget both for SE and for MP (as you can create an instance of it in se using a factory method).
In MP this instance should be injectable.